backend/src/test/java/CustomerDaoTest.java

```diff
@@ -0,0 +1,100 @@
+
+import cz.muni.fi.pa165.pneuservis.backend.PersistenceApplicationContext;
+import javax.inject.Inject;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+import cz.muni.fi.pa165.pneuservis.backend.dao.CustomerDao;
+import cz.muni.fi.pa165.pneuservis.backend.entity.Customer;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+
+
+/***
+ * @author Róbert Ivan , 461468@mail.muni.cz
+ */
+
+
+@ContextConfiguration(classes = PersistenceApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@Transactional
+public class CustomerDaoTest extends AbstractTestNGSpringContextTests{
+    
+  @Inject  
+  private CustomerDao customerManagerDao;
+  
+    @PersistenceContext
+    private EntityManager entityManager;
+    
+    private Customer custumer1;
+    private Customer custumer2;
+    private Customer custumer3;
+    
+    @BeforeMethod
+    public void prepareCustomers() {
+        custumer1 = new Customer("TestName","TestSurname","Testcity","Teststreet","TestzipCode","Testcountry","Testemail","TestphoneNumber");
+        custumer2 = new Customer("TestName","TestSurname","Testcity","Teststreet","TestzipCode","Testcountry","Testemail","TestphoneNumber");
```

Comment:
I'd suggest having all the fields different for each Customer to improve readability.